pkg/controller/srcdst_controller.go

```diff
@@ -105,16 +105,48 @@ func (c *Controller) disableSrcDstIfEnabled(node *v1.Node) {
 }
 
 func (c *Controller) disableSrcDstCheck(instanceID string) error {
-	_, err := c.ec2Client.ModifyInstanceAttribute(
-		&ec2.ModifyInstanceAttributeInput{
-			InstanceId: aws.String(instanceID),
-			SourceDestCheck: &ec2.AttributeBooleanValue{
-				Value: aws.Bool(false),
+	ifaces, err := c.netIfacesNeedingUpdate(instanceID)
```

Comment:
Hmm… seems like we may need to periodically check netiface changes.

From my read of their [docs](https://github.com/aws/amazon-vpc-cni-k8s/blob/master/proposals/cni-proposal.md), it seems like they can have multiple ENIs and multiple IP addresses. Specifically:

> In EC2-VPC, each instance can have multiple elastic network interfaces (ENI) and each ENI can have multiple IPv4 or IPv6 addresses.

and:

> When you create an ENI, you must specify the subnet. Based on the instance type (Limit), each instance can have up to N ENIs and M addresses. One address on each ENI is always marked as primary, while others are secondary addresses.

A quick test of multiple m4.xlarge instances, I'm only seeing 2 ENIs when the k8s node first joins the cluster (out of the [maximum 4 ENIs](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-eni.html#AvailableIpPerENI). This seems to confirm the need to scan for new netifaces.

The naïve pattern is obviously to check every time interval, but I'm sure that won't scale very well with a large number of nodes (not to mention AWS API rate limits). Ideas? Do network interface changes cause new node events?

Reply:
> The naïve pattern is obviously to check every time interval, but I'm sure that won't scale very well with a large number of nodes (not to mention AWS API rate limits). Ideas? Do network interface changes cause new node events?

I'm guessing you would need to run a daemon on each node to watch for new network interfaces (either through OS hooks or by periodically polling) and report back to a coordinator where it would make the API call.